main.go

```diff
@@ -401,49 +401,188 @@ func connectAction(ctx *cli.Context) error {
+	InsightsDisconnected      bool   `json:"insights_disconnected"`
+	InsightsDisconnectedError string `json:"insights_disconnected_error,omitempty"`
+	YggdrasilStopped          bool   `json:"yggdrasil_stopped"`
+	YggdrasilStoppedError     string `json:"yggdrasil_stopped_error,omitempty"`
+	exitCode                  int
+	format                    string
+}
+
+// Error implement error interface for structure DisconnectResult
+func (disconnectResult DisconnectResult) Error() string {
+	var result string
+	switch disconnectResult.format {
+	case "json":
+		data, err := json.MarshalIndent(disconnectResult, "", "    ")
+		if err != nil {
+			return err.Error()
+		}
+		result = string(data)
+	default:
+		result = "error: unsupported document format: " + disconnectResult.format
+	}
+	return result
+}
+
+// printJSONDisconnectResult tries to print the result of disconnect as JSON to stdout.
+// When marshaling of systemStatus fails, then error is returned
+func printJSONDisconnectResult(disconnectResult *DisconnectResult) error {
+	data, err := json.MarshalIndent(disconnectResult, "", "    ")
+	if err != nil {
+		return err
+	}
+	fmt.Println(string(data))
+	return nil
+}
+
+// beforeDisconnectAction ensures the used has supplied a correct `--format` flag
+func beforeDisconnectAction(ctx *cli.Context) error {
+	return setupFormatOption(ctx)
+}
+
+// rhcPrintf is method for printing human-readable output. It suppresses output, when
+// machine-readable format is used.
+func rhcPrintf(format string, a ...interface{}) {
+	if !uiSettings.isMachineReadable {
+		fmt.Printf(format, a...)
+	}
+}
+
+// disconnectAction tries to stop (yggdrasil) rhcd service, disconnect from Red Hat Insights,
+// and finally it unregisters system from Red Hat Subscription Management
 func disconnectAction(ctx *cli.Context) error {
+	var disconnectResult DisconnectResult
+	var machineReadablePrintFunc func(disconnectResult *DisconnectResult) error
+
+	disconnectResult.format = ctx.String("format")
+	format := ctx.String("format")
+	switch format {
+	case "json":
+		machineReadablePrintFunc = printJSONDisconnectResult
+	default:
+		break
+	}
+
+	// When printing of status is requested, then print machine-readable file format
+	// at the end of this function
+	if uiSettings.isMachineReadable {
+		defer func(disconnectResult *DisconnectResult) {
+			// When exit code is zero, then print machine-readable output
+			// When exit code has non-zero value, then disconnectResult is returned as a error
+			if disconnectResult.exitCode == 0 && machineReadablePrintFunc != nil {
+				err := machineReadablePrintFunc(disconnectResult)
+				// When it was not possible to print result of disconnect to machine-readable format, then
+				// change returned error to CLI exit error to be able to set exit code to
+				// a non-zero value
+				if err != nil {
+					panic(fmt.Errorf("unable to print status as %s document: %s", format, err.Error()))
```

Comment:
Typically, it's not necessary to explicitly call Error() on an error type. This
should be sufficient.

```suggestion
					panic(fmt.Errorf("unable to print status as %v document: %v", format, err))
```